lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -2,47 +2,36 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
 import 'package:ui/src/engine.dart';
 import 'package:ui/ui.dart' as ui;
 
 /// Tracks the [FlutterView]s focus changes.
 final class ViewFocusBinding {
-  /// Creates a [ViewFocusBinding] instance.
-  ViewFocusBinding._();
+  ViewFocusBinding(this._viewManager, this._onViewFocusChange);
 
-  /// The [ViewFocusBinding] singleton.
-  static final ViewFocusBinding instance = ViewFocusBinding._();
+  final FlutterViewManager _viewManager;
+  final ui.ViewFocusChangeCallback _onViewFocusChange;
 
-  final List<ui.ViewFocusChangeCallback> _listeners = <ui.ViewFocusChangeCallback>[];
   int? _lastViewId;
   ui.ViewFocusDirection _viewFocusDirection = ui.ViewFocusDirection.forward;
 
-  /// Subscribes the [listener] to [ui.ViewFocusEvent] events.
-  void addListener(ui.ViewFocusChangeCallback listener) {
-    if (_listeners.isEmpty) {
-      domDocument.body?.addEventListener(_keyDown, _handleKeyDown);
-      domDocument.body?.addEventListener(_keyUp, _handleKeyUp);
-      domDocument.body?.addEventListener(_focusin, _handleFocusin);
-      domDocument.body?.addEventListener(_focusout, _handleFocusout);
-    }
-    _listeners.add(listener);
-  }
+  StreamSubscription<int>? _onViewCreatedListener;
 
-  /// Removes the [listener] from the [ui.ViewFocusEvent] events subscription.
-  void removeListener(ui.ViewFocusChangeCallback listener) {
-    _listeners.remove(listener);
-    if (_listeners.isEmpty) {
-      domDocument.body?.removeEventListener(_keyDown, _handleKeyDown);
-      domDocument.body?.removeEventListener(_keyUp, _handleKeyUp);
-      domDocument.body?.removeEventListener(_focusin, _handleFocusin);
-      domDocument.body?.removeEventListener(_focusout, _handleFocusout);
-    }
+  void init() {
+    domDocument.body?.addEventListener(_keyDown, _handleKeyDown);
+    domDocument.body?.addEventListener(_keyUp, _handleKeyUp);
+    domDocument.body?.addEventListener(_focusin, _handleFocusin);
+    domDocument.body?.addEventListener(_focusout, _handleFocusout);
+    _onViewCreatedListener = _viewManager.onViewCreated.listen(_handleViewCreated);
   }
 
-  void _notify(ui.ViewFocusEvent event) {
-    for (final ui.ViewFocusChangeCallback listener in _listeners) {
-      listener(event);
-    }
+  void dispose() {
+    domDocument.body?.removeEventListener(_keyDown, _handleKeyDown);
+    domDocument.body?.removeEventListener(_keyUp, _handleKeyUp);
+    domDocument.body?.removeEventListener(_focusin, _handleFocusin);
+    domDocument.body?.removeEventListener(_focusout, _handleFocusout);
```

Comment:
Can we attach focusin/focusout to the rootElements of the registered views? Wouldn't that remove a lot of the gymnastics we need to do to see if the focus event comes from a flutter view or not? (the answer will always be: "yes")